lib/mongo/error/wait_queue_timeout.rb

```diff
@@ -0,0 +1,41 @@
+# Copyright (C) 2014-2019 MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Mongo
+  class Error
+
+    # Exception raised if a timeout occurs when attempting to acquire a connection from a pool.
+    #
+    # @since 2.7.0
+    class WaitQueueTimeout < Error
+
+      # @return [ String ] address The address of the server the pool's connections connect to.
+      #
+      # @since 2.7.0
+      attr_reader :address
+
+      # Instantiate the new exception.
+      #
+      # @example Instantiate the exception.
+      #   Mongo::Error::WaitQueueTimeout.new(address)
+      #
+      # @since 2.7.0
+      def initialize(address)
+        @address = address
+
+        super("timeout when attempting to check out a connection from pool with address #{address}")
```

Comment:
Per https://github.com/mongodb/mongo-ruby-driver/pull/1178, this message should include also the size of the pool and the timeout that was exceeded, and should begin with a capital letter.

Reply:
Done